### ISSUE TITLE
Bug: no-unused-private-class-members does not detect usage in static methods

### ISSUE THREAD

**Issue by MiK546** [typescript-eslint/typescript-eslint#12189]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=MYGwhgzhAEAqCmEAuBvAUNaAHATgSwDcwl5od4wATAewDsQBPaS4sARgC5pl9aBzANxoM2fERLRgdHgFdgSajgAULJOy488-AJTpMmJAAs8EAHSr20ALzNWbIZgC%2BwzFhkAjEHmDc1Sb9CQDLQ%2BtPAA7ip2Gki8fNpcAAo41AC2JvAAPAjIAHx6%2BlK0yNAkyACSxWohpDZh4XCISFFqbNoO%2BmDhYHhIpU2VyGA15tQAghDBwJW9eGBeEMR4dErtIpjkSDI4tP0VVcPA8B3OIriExKRBIczjkyEz-vMmSyu665LS1CDwpiDUfCUACJyqkwHwtKRwqQaNw0qQ8KksIpqn1rj4tLNnot-HRoIZ4OQgWt9J9it9fv9AUYTOY7CToM5HEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtYlQBuAQ3yJaZeOOTJaAW0RKARomioMkTdA7RI4MAF8QJoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Test{
  private readonly data1: string;

  private constructor(data1: string){
    this.data1 = data1;
  }

  public static async new(data1: string): Promise<Test>{
    const testInstance = new Test(data1);
    await testInstance.doAsyncInitialisation();
    return testInstance;
  }

  private async doAsyncInitialisation(){
    console.log("Imagine we do some important async initialisation here");
    console.log(this.data1);
  }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unused-private-class-members": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

In the example code `doAsyncInitialisation()` should not be marked as unused.

### Actual Result

In the example code `doAsyncInitialisation()` is marked as unused.

### Additional Info

I know the rule has some limitations listed in the documentation so maybe this falls under those but as the instance the method is called from is correctly typed at compile time hopefully this is a valid bug.

**Comment by bradzacher** [comment]
This is limitation number 2.
https://typescript-eslint.io/rules/no-unused-private-class-members/#limitations

You're accessing the property via an instantiation. We cannot detect this.
